### ISSUE TITLE
AutoLink should use link command

### ISSUE THREAD

**Issue by scofalik** [ckeditor/ckeditor5#7909]
## 📝 Provide detailed reproduction steps (if any)

`AutoLink` plugin does not use link command? Why? cc @jodator 

It would be nice if it could use link command. At the moment there is a very minor bug in TC related to that. But in general, if we would have some additional logic in link command, we may forget about applying it in autolinking. So, current implementation seems to be "dangerous".

One problem that comes to my mind is that we would have to mangle with selection :roll\_eyes: . Soo.. when do we rewrite all the commands so that they are not depended on selection? :>

**Comment by jodator** [comment]
We probably should gather all commands requirements and see what should be done.

At this moment `LinkCommand` does not accept range and it's API outstands from the newer ones.

**Comment by Reinmar** [comment]
Blocked by https://github.com/ckeditor/ckeditor5/issues/1954.

**Comment by ejwill04** [comment]
@jodator - I don't mean to hijack this issue if this should be a separate issue, but I wanted to bring up something that seems related.

If I have `AutoLink` enabled and also any other `Link` config properties such as, `addTargetToExternalLinks`, `decorators`, or `defaultProtocol`, the `AutoLink` plugin does not trigger these actions.  

A simple example would be to add both `AutoLink` and `defaultProtocol: 'http://'`, then type a link (`www.google.com`).  The link gets created by the `AutoLink`, but the link that's created doesn't include the defaultProtocol, which causes the link to be treated as a relative link.

**Comment by jodator** [comment]
@ejwill04 You're right - we do not add `http://` to the auto-linked link. I'd suspect that writing `http://www.google.com` would add the target as [automatic decorators](https://ckeditor.com/docs/ckeditor5/latest/features/link.html#adding-attributes-to-links-based-on-predefined-rules-automatic-decorators) are applied during conversion from the model to the view (ie. data). So here in your case it would not pick `www.google.com` as such... but it could probably. The [manual decorators](https://ckeditor.com/docs/ckeditor5/latest/features/link.html#adding-attributes-to-links-using-the-ui-manual-decorators) are applied by the UI.

However, the `defaultProtocol` configuration option is used in the UI only. I think that this might be used for auto-link probably as well as the use-case is pretty similar but OTOH we should think that through as I'm unsure whether this should be tied or should we have an option to disable it for auto-link or UI only. I'll report separate issue for this.

**Comment by CKEditorBot** [comment]
There has been no activity on this issue for the past year. We've marked it as stale and will close it in 30 days. We understand it may still be relevant, so if you're interested in the solution, leave a comment or reaction under this issue.

**Comment by CKEditorBot** [comment]
We've closed your issue due to inactivity over the last year. We understand that the issue may still be relevant. If so, feel free to open a new one (and link this issue to it).
